src/Chapter1.hs

```diff
@@ -537,7 +542,12 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+  | (x > y && z > x) || (x > z && y > x) = x
+  | (y > x && z > y) || (y > z && x > y) = y
+  | (z > x && y > z) || (z > y && x > z) = z
+  | otherwise = x
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = x`, to tell the compiler, that your pattern matching is exhaustive 🙂 But as you can notice, you can just collapse the last two guards and return `z` in the last option.